cdap-ui/app/cdap/services/helpers.js

```diff
@@ -282,6 +283,38 @@ const composeEnhancers = (storeTitle) =>
       name: storeTitle
     }) : compose;
 
+const convertMapToKeyValuePairsObj = (obj) => {
+  let keyValuePairsObj = {};
+  keyValuePairsObj.pairs = Object.keys(obj).map(objKey => {
+    return {
+      key: objKey,
+      value: obj[objKey],
+      uniqueId: 'id-' + uuid()
+    };
+  });
+  if (!keyValuePairsObj.pairs.length) {
+    keyValuePairsObj.pairs.push({
+      key: '',
+      value: '',
+      uniqueId: 'id-' + uuid()
+    });
+  }
+  return keyValuePairsObj;
+};
+
+const convertKeyValuePairsObjToMap = (keyValues) => {
+  let map = {};
+  if (keyValues.pairs) {
+    keyValues.pairs.forEach((currentPair) => {
+      if (currentPair.key.length > 0 && currentPair.value.length > 0) {
+        let key = currentPair.key;
+        map[key] = currentPair.value;
+      }
+    });
+  }
+  return map;
+};
```

Comment:
I think these function should reside inside the KeyValuePairs component, instead of helpers. This is not generic enough to be a shared helper